src/ows/src/main/java/org/geoserver/ows/Dispatcher.java

```diff
@@ -1771,6 +1771,18 @@ void handleServiceException( ServiceException se, Service service, Request reque
             handler = new SOAPServiceExceptionHandler(handler);
         }
 
+        // remove content disposition if set (to allow browsers inline display of exception)
+        HttpServletResponse httpResponse = request.getHttpResponse();
+        if (httpResponse.containsHeader(HttpHeaders.CONTENT_DISPOSITION)) {
```

Comment:
Looks like a legit change, but not sure were this is coming from.

Reply:
When requesting a downloadable file (like shapefile) the content type and content disposition are set before giving the output stream to the respnse. On exception, they are still there, and the browser proposes to download what looks like a zip file, while it's really an exception. This causes some confusion on Linux, but much more so on OSs uses the extension to determine the type of application needed to open the file. So this change tries to reset the content disposition if set to avoid that.